src/overlays/actors/ovl_Bg_Kin2_Bombwall/z_bg_kin2_bombwall.c

```diff
@@ -28,50 +40,194 @@ const ActorInit Bg_Kin2_Bombwall_InitVars = {
+
+    for (i = 0, temp_s3 = 0; i < 6; i++) {
+        temp_a0 = (i + 1) * 15.f;
+        for (j = 0; j < ARRAY_COUNT(D_80B6E730); j++) {
+            temp_s3++;
+            temp_s3 &= 7;
+
+            spD8.x = D_80B6E730[j] + (s32)(((u32)Rand_Next()) >> 0x1C);
+            spD8.y = ((Rand_ZeroOne() - 0.5f) * 15.0f) + temp_a0;
+            spD8.z = (Rand_ZeroOne() * 20.0f) - 10.0f;
+
+            spCC.x = (2.0f * (Rand_ZeroOne() - 0.5f)) + (spD8.x * 0.018461538f);
+            spCC.y = (Rand_ZeroOne() * 7.0f) + 4.0f;
+            spCC.z = spD8.z * 0.3f;
+
+            Matrix_MultVec3f(&spD8, &pos);
+            Matrix_MultVec3f(&spCC, &velocity);
+
+            pos.x += this->dyna.actor.world.pos.x;
+            pos.y += this->dyna.actor.world.pos.y;
+            pos.z += this->dyna.actor.world.pos.z;
+
+            if (Rand_Next() % 4 == 0) {
+                phi_s0 = 0x20;
+            } else {
+                phi_s0 = 0x40;
+            }
+
+            if (temp_s3 < 2 || Rand_Next() > 0) {
+                phi_s0 |= 1;
+                phi_s1 = 1;
+                func_800B0E48(globalCtx, &pos, &gZeroVec3f, &D_80B6E724, &D_80B6E71C, &D_80B6E720,
+                              (((u32)Rand_Next() >> 0x1B) + 0x46),
+                              ((((u32)Rand_Next()) >> 0x1A) + 0x3C)); // for dust spawn
+            } else {
+                phi_s1 = 0;
+            }
+            EffectSsKakera_Spawn(globalCtx, &pos, &velocity, &pos, -0x226, phi_s0, 0x1E, 0, 0, D_80B6E738[temp_s3],
+                                 phi_s1, 0, 0x32, -1, 0x1F5, D_06000128);
+        }
+    }
+}
+
+void BgKin2Bombwall_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgKin2Bombwall* this = (BgKin2Bombwall*)thisx;
+    ColliderCylinder* bombwallCollider;
+
+    Actor_ProcessInitChain(&this->dyna.actor, D_80B6E748);
+    DynaPolyActor_Init((DynaPolyActor*)this, 0);
+    bombwallCollider = &this->collider;
+    Collider_InitCylinder(globalCtx, bombwallCollider);
+    if (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x7F) != 0) {
+        Actor_MarkForDeath(&this->dyna.actor);
+        return;
+    }
+    DynaPolyActor_LoadMesh(globalCtx, (DynaPolyActor*)this, (CollisionHeader*)&D_06000490);
+    Collider_SetCylinder(globalCtx, bombwallCollider, &this->dyna.actor, &D_80B6E6F0);
+    Collider_UpdateCylinder(&this->dyna.actor, bombwallCollider);
+    Actor_SetFocus(&this->dyna.actor, 60.0f);
+    BgKin2Bombwall_SetupWait(this);
+}
+
+void BgKin2Bombwall_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgKin2Bombwall* this = (BgKin2Bombwall*)thisx;
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void BgKin2Bombwall_SetupWait(BgKin2Bombwall* arg0) {
+    arg0->actionFunc = BgKin2Bombwall_Wait;
+}
+
+void BgKin2Bombwall_Wait(BgKin2Bombwall* arg0, GlobalContext* arg1) {
+    if (func_80B6E020(arg0, arg1) != 0) { // checks if AC collision happened
```

Comment:
Can remove comment once function named, it returns boolean so do not need the != 0.